packages/beacon-node/src/network/processor/gossipQueues/index.ts

```diff
@@ -88,8 +87,8 @@ const indexedGossipQueueOpts: {
   [GossipType.beacon_attestation]: {
     maxLength: 24576,
     indexFn: (item: PendingGossipsubMessage) => {
-      const {topic, msg} = item;
-      return getSeenAttDataKey(ForkSeq[topic.fork], msg.data);
+      // Note indexFn is fork agnostic despite changes introduced in Electra
```

Comment:
extracting attestation data for all forks is fine here but need to enhance the gossip attestation validation because this key is currently used for the cache over there https://github.com/ChainSafe/lodestar/blob/68cefffed48c4912227708b7233f6abfe4a0df15/packages/beacon-node/src/chain/validation/attestation.ts#L276

Reply:
> extracting attestation data for all forks is fine here but need to enhance the gossip attestation validation because this key is currently used for the cache over there

Can you elaborate what kind of enhancement is needed for validation? `seenAttestationKey` for cache is still fork-dependent. Gossip queue batching is fork-independent.

Reply:
@ensi321  nvm, the code works but does not go with the comment. `seenAttestationKey` is expected from gossip validation logic but gossipHandlers passed in `attDataBase64` instead

also `getSeenAttDataKeyFromSignedAggregateAndProofElectra()` is not correct because `toBase64(Buffer.concat[a, b])` is not the same to `toBase64(a) + toBase64(b)`

I fixed them in #7009 please have look